metadata/client.go

```diff
@@ -572,24 +597,29 @@ func (def TrainingSetDef) ResourceType() ResourceType {
 	return TRAINING_SET_VARIANT
 }
 
-func (def TrainingSetDef) Serialize() *pb.TrainingSetVariant {
-	return &pb.TrainingSetVariant{
-		Name:        def.Name,
-		Variant:     def.Variant,
-		Description: def.Description,
-		Owner:       def.Owner,
-		Provider:    def.Provider,
-		Status:      &pb.ResourceStatus{Status: pb.ResourceStatus_CREATED},
-		Label:       def.Label.Serialize(),
-		Features:    def.Features.Serialize(),
-		Schedule:    def.Schedule,
-		Tags:        &pb.Tags{Tag: def.Tags},
-		Properties:  def.Properties.Serialize(),
+func (def TrainingSetDef) Serialize(ctx context.Context) *pb.TrainingSetVariantRequest {
```

Comment:
Don't pass in the context, pass in the requestID

Reply:
Done, but why?